### ISSUE TITLE
coverage workflow not working (for linter)

### ISSUE THREAD

**Issue by pq** [dart-lang/coverage#382]
See for example:

https://github.com/dart-lang/linter/runs/6414611200?check_suite_focus=true

```
Generating LCOV report...
Unhandled exception:
FormatException: Unexpected character (at character 1)
# This file is deprecated. Tools should instead consume 
^

#0      throwError (package:package_config/src/errors.dart:32:35)
#1      parsePackageConfigString (package:package_config/src/package_config_json.dart:54:12)
#2      PackageConfig.parseString (package:package_config/src/package_config.dart:94:7)
#3      Resolver._parsePackages (package:coverage/src/resolver.dart:115:23)
#4      Resolver.create (package:coverage/src/resolver.dart:34:13)
#5      HitMap.parseJson (package:coverage/src/hitmap.dart:45:37)
#6      HitMap.parseFiles (package:coverage/src/hitmap.dart:173:41)
#7      main (file:///home/runner/.pub-cache/hosted/pub.dartlang.org/coverage-1.3.1/bin/format_coverage.dart:66:31)
#8      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:295:32)
#9      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:192:12)
```

Our workflow is here:

https://github.com/dart-lang/linter/blob/master/.github/workflows/build.yaml#L46

Any help appreciated!

@liamappelbe ?

**Comment by liamappelbe** [comment]
Culprit is this line: https://github.com/dart-lang/linter/blob/e76b93520b5f9d6b045fa5334a51ccd741024250/tool/bot.sh#L77

The .packages file is deprecated. The new flag you should use is `--package`, and set it to the root dir of your package. The default value of this flag is `.`, so you can probably just delete the `--packages` flag from the format command and this default package dir should take over.

But I just added a new tool that should simplify bot.sh. You can try migrating to `test_with_coverage`, which runs the tests, and collects and formats the coverage. You should be able to replace the entire `elif [ "$LINTER_BOT" = "coverage" ]; then` with a one liner.

**Comment by albertodev01** [comment]
I am having the same issue here. I have upgraded to `1.3.1` and I get this error:

![error](https://user-images.githubusercontent.com/2848696/168182448-182d3d4d-403a-4e7c-8535-a3fe8abf8646.png)

CI is also failing due to this error

**Comment by liamappelbe** [comment]
Yeah, try rerunning that command without the `--packages` flag

**Comment by albertodev01** [comment]
There should have been an explicit note in the changelog to remove that flag 🙂

**Comment by liamappelbe** [comment]
The changelog mentions that the `--packages` flag is deprecated, but maybe that could have been more clear.

You might want to migrate away from calling those commands manually in your CI though, and using test_with_coverage instead. I just migrated one of my packages. It was easy and eliminated the need for a custom shell script. This is what the migration looks like: https://github.com/liamappelbe/fftea/pull/9

**Comment by pq** [comment]
Fantastic.  Thanks so much for the assist!

<img width="856" alt="image" src="https://user-images.githubusercontent.com/67586/168197339-15e08e38-682a-4d05-aee2-3e0caacce7c7.png">

**Comment by devoncarew** [comment]
You might also consider shipping a version, which requires dart 2.17 (where pub was removed), still accepts the `--packages` flag, but immediately fails when it's passed w/ a message about switching to the `--package` flag. I think the errors that people will get will be much more actionable (with the drawback that new versions of coverage won't be shipped to people on sdk before 2.17).
